react/src/pages/datasets/DatasetTable.tsx

```diff
@@ -85,9 +85,23 @@ export default function DatasetTable() {
             />
             {infoDataset && (
                 <DatasetInfoDialog
-                    dataset_id={infoDataset.dataset_id}
+                    dataset={infoDataset}
                     open={showInfo}
-                    onClose={() => setShowInfo(false)}
+                    onClose={() => {
+                        setShowInfo(false);
+                    }}
+                    onUpdate={(dataset_id: string, newDataset: { [key: string]: any }) => {
+                        const updatedDatasets = [...datasets];
+                        const idx = datasets.findIndex(
+                            dataset => dataset.dataset_id === dataset_id
+                        );
+                        const updatedDataset = { ...datasets[idx] } as { [key: string]: any };
+                        Object.keys(newDataset).forEach(key => {
+                            updatedDataset[key] = newDataset[key];
+                        });
+                        updatedDatasets[idx] = updatedDataset as Dataset;
+                        setDatasets(updatedDatasets);
```

Comment:
This can be done with a single `map` call over `datasets` and passing the result to `setDatasets`. The only element that should be replaced by map is the one with matching `dataset_id`, where we make a copy of the old and mixin properties of the new.